utils/build-script-impl

```diff
@@ -712,6 +712,18 @@ function set_build_options_for_host() {
           -DLLVM_TOOL_COMPILER_RT_BUILD:BOOL="$(false_true ${SKIP_BUILD_COMPILER_RT})"
           -DLLVM_BUILD_EXTERNAL_COMPILER_RT:BOOL="$(false_true ${SKIP_BUILD_COMPILER_RT})"
       )
+      # Android doesn't support building all of compiler-rt yet.
+      if [[ "${host}" == "android-"* ]]; then
+        llvm_cmake_options+=(
+            -DCOMPILER_RT_INCLUDE_TESTS:BOOL=FALSE
+        )
+      fi
+    fi
+
+    if [[ "${host}" == "android-"* ]]; then
+      llvm_cmake_options+=(
+        -DDEFAULT_SYSROOT:STRING="$(dirname ${PREFIX})"
```

Comment:
Minor: `:PATH` instead of `:STRING`?

This one might also deserve a small comment, since, for some reason, it is not prefixed with `LLVM_` or `CLANG_` and it is not clear that it actually changes the default sysroot used by the compiled `clang` (IIUC).

Also, I cannot find where `PREFIX` is coming from. I see it used in other case in `build-script-impl` (added by you in #29296). Are you passing `PREFIX` as an environmental variable in your scripts and this script is picking it up?

If that's the case, I don't know if that's a very "clean" way of doing things, but I would recommend using a more descriptive name than just `PREFIX`, and maybe add some information about it, so people do not need to spelunk into the code for its meaning.

Reply:
Yes, `PREFIX` is set by the Termux build scripts when building the Swift toolchain for Android, I will add a comment explaining.

Reply:
Added the comment explanations and I just checked: [LLVM wants a `STRING`](https://github.com/apple/llvm-project/blob/aafb3662103f4b3df315967c0cf4f6eec6bff0c4/clang/CMakeLists.txt#L237), so that's what I give it.